internal/configs/virtualserver_test.go

```diff
@@ -3834,3 +3870,378 @@ func TestIsTLSEnabled(t *testing.T) {
 	}
 
 }
+
+func TestGenerateRewrites(t *testing.T) {
+	tests := []struct {
+		path         string
+		proxy        *conf_v1.ActionProxy
+		internal     bool
+		originalPath string
+		expected     []string
+	}{
+		{
+			proxy:    nil,
+			expected: nil,
+		},
+		{
+			proxy: &conf_v1.ActionProxy{
+				RewritePath: "",
+			},
+			expected: nil,
+		},
+		{
+			path: "/path",
+			proxy: &conf_v1.ActionProxy{
+				RewritePath: "/rewrite",
+			},
+			expected: nil,
+		},
+		{
+			path:     "/_internal_path",
+			internal: true,
+			proxy: &conf_v1.ActionProxy{
+				RewritePath: "/rewrite",
+			},
+			originalPath: "/path",
+			expected:     []string{`^ $request_uri`, `"^/path(.*)$" "/rewrite$1" break`},
+		},
+		{
+			path:     "/_internal_path",
+			internal: true,
+			proxy: &conf_v1.ActionProxy{
+				RewritePath: "/rewrite",
+			},
+			originalPath: "/path",
+			expected:     []string{`^ $request_uri`, `"^/path(.*)$" "/rewrite$1" break`},
+		},
+		{
+			path:     "~/regex",
+			internal: true,
+			proxy: &conf_v1.ActionProxy{
+				RewritePath: "/rewrite",
+			},
+			originalPath: "/path",
+			expected:     []string{`^ $request_uri`, `"^/path(.*)$" "/rewrite$1" break`},
+		},
+		{
+			path:     "~/regex",
+			internal: false,
+			proxy: &conf_v1.ActionProxy{
+				RewritePath: "/rewrite",
+			},
+			expected: []string{`"^/regex$" "/rewrite" break`},
```

Comment:
this might be a bug in the `generateRewrites` . why are we appending `$`? 

Consider the following config:
```nginx
    location ~/test {
        rewrite ^/test$ "/rewrite" break;
        proxy_pass http://test;
    }
```
In this case:
* `/test` -> `/rewrite` - OK
* `/test/abc` -> `/test/abc` :( 

If we step back and go to CRDs, in this case it is a responsibility of the user that the path and rewrite work together. So they should be something like  `~ /tea/?(.*)` and `/$1` (from the example)

Reply:
Yes you're right, I was adding a `$` in the regex case too by mistake, as you say, this should be responsibility of the user. I'll sync up with @vepatel to make sure we cover all the use cases in the tests. Thanks for the heads up.